trin-core/src/types/json_helpers.rs

```diff
@@ -0,0 +1,416 @@
+use crate::types::block_body::{AccessList, AccessListItem, StorageKeys};
+use crate::types::receipts::{LogEntry, TransactionId, TransactionOutcome};
+use crate::utils::bytes::hex_decode;
+use anyhow::anyhow;
+use bytes::Bytes;
+use ethereum_types::{Bloom, H160, H256, U256};
+use serde_json::{Map, Value};
+
+//
+// Custom util fns for 0x-prefixed hexstrings returned by jsonrpc
+//
+
+pub fn convert_to_vec_value(val: &Value) -> anyhow::Result<&Vec<Value>> {
+    val.as_array()
+        .ok_or_else(|| anyhow!("Value is not an array."))
+}
+
+fn convert_to_str(val: &Value) -> anyhow::Result<&str> {
+    val.as_str()
+        .ok_or_else(|| anyhow!("Value is not a string."))
+}
+
+fn convert_to_obj(val: &Value) -> anyhow::Result<&Map<String, Value>> {
+    val.as_object()
+        .ok_or_else(|| anyhow!("Value is not an object."))
+}
+
+pub struct JsonVecH256(Vec<H256>);
+
+impl TryFrom<&Value> for JsonVecH256 {
+    type Error = anyhow::Error;
+
+    fn try_from(val: &Value) -> anyhow::Result<Self> {
+        let result = convert_to_vec_value(val)?;
+        let mut hashes = vec![];
+        for val in result {
+            let hash = JsonH256::try_from(val)?.into();
+            hashes.push(hash);
+        }
+        Ok(Self(hashes))
+    }
+}
+
+#[allow(clippy::from_over_into)]
```

Comment:
The goal of these conversion functions is to go from `Value` -> `Primitive` (`u8`, `H256`, etc.). This requires some use of a wrapped type. If we listen to clippy and go with `impl From<Value> for Primitive` this will require more layers of wrapped types, eg. `impl From<WrappedValue> for WrappedPrimitive`, imo it's much preferable to minimize the amount of wrapped types we use and just ignore clippy in this one case.